### PR TITLE
Add site options for WPCOM + GitHub integration

### PR DESCRIPTION
**Pull request by mpkelly**
Related https://github.com/Automattic/wp-calypso/issues/70380

## Proposed changes:

* Add three new site options required for the WPCOM + GitHub integration. 
  - `wpcom_github_hosting_repo_name`
  - `wpcom_github_hosting_branch_name`
  - `wpcom_github_hosting_base_path`

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Follow the steps to apply this patch to your sandbox by using the instructions below in a separate comment. 
* Go to `/hosting-config/:site` for a site where you have [connected GitHub](https://github.com/Automattic/wp-calypso/pull/73030) and you should see the values that were saved when you hit "Connect" if you check the `sites.options` in the response of `/rest/v1.2/sites/:site`. 
* Review the code: Did I add the code to the correct place? Do we even need a change log entry?

### CODE REVIEW

File: projects/plugins/jetpack/sal/class.json-api-site-base.php [hunk 1]
```diff
@@ -1458,4 +1458,26 @@ public function get_launchpad_checklist_tasks_statuses() {
...
+	public function get_github_hosting_branch_name() {
+		return get_option( 'github_hosting_branch_name', '' );
+	}
+
+	/**
+	 * Get the site option for the base path to use when copying
+	 * files from the connected GitHub repo to the Atomic file system.
+	 */
+	public function get_github_hosting_base_path() {
+		return get_option( 'github_hosting_base_path', '' );
```
Comment (jeherve) [inline projects/plugins/jetpack/sal/class.json-api-site-base.php]: Is it too late to change those option names, and prefix them with `wpcom_` or `jetpack_`? I worry that such option names will be used by other plugins in the WordPress.org directory, and would conflict with our implementation in the future.

**Review by jeherve — COMMENTED** [review]
This looks good, but I do worry about the non-prefixed option names. Is it too late to change them?

File: projects/plugins/jetpack/sal/class.json-api-site-base.php (hunk 1 above)
Comment (zaguiini) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: Let's prefix them with `wpcom_` 👍

Comment (mpkelly) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: I've updated the PR to use the `wpcom` prefix.

**Review by jeherve — APPROVED** [review]

**Comment by p-jackson** [comment]
@mpkelly Rather than expose these option via the API we share with Jetpack I think it makes more sense to have our own GitHub-deployment-specific endpoint D102157-code

**Comment by zaguiini** [comment]
As explained by @p-jackson, it makes more sense to expose the data in a specific endpoint rather than adding it to the site options object.

**Pull request closed without merging**